src/__test__/client.test.ts

```diff
@@ -0,0 +1,213 @@
+    const toLock = { txid: utxos[0].txid, vout: utxos[0].vout }
+    await client.lockUnspent(false, [toLock])
+    let lockList = await client.listLockUnspent()
+    expect(lockList).toContainEqual(toLock)
+
+    await client.lockUnspent(true)
+
+    lockList = await client.listLockUnspent()
+    expect(lockList.length).toEqual(0)
+  })
+  test('importaddress', async () => {
+    const address = 'bcrt1qlgmznucxpdkp5k3ktsct7eh6qrc4tju7ktjukn'
+    await client.importAddress(address)
+    const info = await client.getAddressInfo(address)
+    expect(info.iswatchonly).toBeTruthy()
+  })
+  test('importpublickey', async () => {
+    const privkey =
+      '0000000000000000000000000000000000000000000000000000000000000001'
+    const pubkey = cfdjs.GetPubkeyFromPrivkey({ privkey, isCompressed: true })
+      .pubkey
+    const address = cfdjs.CreateAddress({
+      keyData: { type: 'pubkey', hex: pubkey },
+      network: 'regtest',
+      hashType: 'p2wpkh',
+    }).address
+    await client.importPubKey(pubkey)
+    const info = await client.getAddressInfo(address)
+    expect(info.iswatchonly).toBeTruthy()
+  })
+  test('createWallet', async () => {
+    const walletParams = [
+      { name: 'alice' },
+      { name: 'bob', disablePrivateKeys: true },
+      { name: 'carol', disablePrivateKeys: false, blank: true },
+      {
+        name: 'denise',
+        disablePrivateKeys: false,
+        blank: false,
+        passphrase: 'pass',
+      },
+      {
+        name: 'emily',
+        disablePrivateKeys: false,
+        blank: false,
+        passphrase: '',
+        avoid_reuse: true,
+      },
+    ]
+
+    for (const params of walletParams) {
+      const result = await client.createWallet(
+        params.name,
+        params.disablePrivateKeys,
+        params.blank,
+        params.passphrase,
+        params.avoid_reuse
+      )
+
+      expect(result.name).toEqual(params.name)
+      const expectedWarning =
+        params.passphrase === undefined || params.passphrase === ''
+          ? 'Empty string given as passphrase, wallet will not be encrypted.'
+          : ''
+
+      expect(result.warning).toEqual(expectedWarning)
+
+      const walletClient = new Client({
+        baseURL: url + '/wallet/' + params.name,
+        auth,
+      })
+
+      const info = await walletClient.getWalletInfo()
+      expect(info.walletname).toEqual(params.name)
+      expect(info.private_keys_enabled).toEqual(!params.disablePrivateKeys)
+      expect(info.hdseedid === undefined).toEqual(
+        params.disablePrivateKeys || false || params.blank || false
+      )
+      expect(info.avoid_reuse).toEqual(params.avoid_reuse || false)
+      expect(info.scanning).toEqual(false)
+    }
+  })
+  test('(un)lockwallet', async () => {
+    const walletName = 'testlock'
+    const walletPass = 'pass'
+
+    await client.createWallet(walletName, undefined, undefined, walletPass)
+    const walletClient = new Client({
```

Comment:
what about creating a new method and child `getClientWallet(name: string):ClientWallet` from the default client ?
as I suppose some methods like `walletPassphrase(walletPass, 10)` should only be called with a `ClientWallet`

Reply:
It's not that simple. If you don't create any wallet, then you can use the default wallet with the base url. But as soon as you create a wallet, you need to start using `/wallet` for default wallet and `/wallet/x` for wallet x. I would rather leave this responsibility to the caller and keep things simple.

Reply:
I see, didn't think about the default wallet.
but what about the url ?, don't really like to give it like this as it is no more a baseURL, it kind of changes depending on the contents.
Should you list the wallets first ? and maybe provide  a `switchWallet` method

Reply:
> don't really like to give it like this as it is no more a baseURL, it kind of changes depending on the contents.

Not sure what you mean. I can change the name of the parameter if you want (just used baseURL to match with axios).

> Should you list the wallets first ? and maybe provide a switchWallet method

Mmm I'm hesitant on this one. On the one hand it's true that it could be a nice feature, but on the other hand it increases complexity of the library. For now I think I would rather keep things simple and leave the responsibility to the caller.